howso/direct/client.py

```diff
@@ -407,16 +408,18 @@ def _auto_persist_trainee(self, trainee_id: str):
         trainee_id : str
             The ID of the Trainee to persist.
         """
-        try:
-            trainee = self.trainee_cache.get(trainee_id)
-            if trainee.persistence == 'always':
-                self.amlg.store_entity(
-                    handle=trainee_id,
-                    file_path=self.resolve_trainee_filepath(trainee_id)
-                )
-        except KeyError:
-            # Trainee not cached, ignore
-            pass
+        trainee = self.trainee_cache.get(trainee_id)
```

Comment:
trainee_cache is not a dict, calling get() without a default value will raise KeyError.

```suggestion
        trainee = self.trainee_cache.get(trainee_id, None)
```